src/parser/hevc_parser.cpp

```diff
@@ -1772,19 +1783,23 @@ ParserResult HevcVideoParser::ParseSliceHeader(uint8_t *nalu, size_t size, HevcS
                 }
                 if ((p_slice_header->collocated_from_l0_flag && p_slice_header->num_ref_idx_l0_active_minus1 > 0) || (!p_slice_header->collocated_from_l0_flag && p_slice_header->num_ref_idx_l1_active_minus1 > 0)) {
                     p_slice_header->collocated_ref_idx = Parser::ExpGolomb::ReadUe(nalu, offset);
+                    CHECK_ALLOWED_MAX(p_slice_header->collocated_ref_idx, p_slice_header->num_ref_idx_l0_active_minus1);
```

Comment:
Should check against num_ref_idx_l0_active_minus1 or num_ref_idx_l1_active_minus1 according to slice type. As per  spec:
When slice_type is equal to P or when slice_type is equal to B and collocated_from_l0_flag is equal to 1,
collocated_ref_idx refers to a picture in list 0, and the value of collocated_ref_idx shall be in the range of 0 to
num_ref_idx_l0_active_minus1, inclusive.
When slice_type is equal to B and collocated_from_l0_flag is equal to 0, collocated_ref_idx refers to a picture in list 1
and the value of collocated_ref_idx shall be in the range of 0 to num_ref_idx_l1_active_minus1, inclusive.

Reply:
Yes. We need to definitely add more error checks. I just added some trivial ones

Reply:
Here is the code to replace the above line:

if (p_slice_header->slice_type == HEVC_SLICE_TYPE_P || (p_slice_header->slice_type == HEVC_SLICE_TYPE_B && p_slice_header->collocated_from_l0_flag == 1)) {
                        CHECK_ALLOWED_MAX(p_slice_header->collocated_ref_idx, p_slice_header->num_ref_idx_l0_active_minus1);
                    } else if (p_slice_header->slice_type == HEVC_SLICE_TYPE_B && p_slice_header->collocated_from_l0_flag == 0) {
                        CHECK_ALLOWED_MAX(p_slice_header->collocated_ref_idx, p_slice_header->num_ref_idx_l1_active_minus1);
                    }

Reply:
With the above two requested changes, all HEVC conformance streams are now passing.